agents/go-agents/core/event/event.go

```diff
@@ -81,19 +82,20 @@ func (bus *Bus) Pub(e E) {
 func (bus *Bus) Sub(consumer Consumer, eType string) {
 	bus.Lock()
 	defer bus.Unlock()
-	bus.consumers[eType] = append(bus.consumers[eType], consumer)
+	bus.consumers[eType] = append(bus.copyConsumers(eType), consumer)
 }
 
 // SubAny does the same as Sub func, but for multiple event types.
 func (bus *Bus) SubAny(consumer Consumer, types ...string) {
 	bus.Lock()
 	defer bus.Unlock()
 	for _, t := range types {
-		bus.consumers[t] = append(bus.consumers[t], consumer)
+		bus.consumers[t] = append(bus.copyConsumers(t), consumer)
 	}
```

Comment:
Since the whole function is covered by a lock do we really need to copy consumers before appending the new one?